docs/source/add_new_model.md

```diff
@@ -0,0 +1,573 @@
+
+In the same way, the Hugging Face team will open comments when reviewing your code. We recommend asking most questions on GitHub on your PR. For some very general
+questions that are not very useful for the public, feel free to ping the Hugging Face team by Slack or mail.
+
+#### Adapt the generated model's code for brand_new_bert
+	
+At first, we will focus only on the model itself and not care about the tokenizer. All the relevant code should be found in 
+the generated files `src/transformers/models/brand_new_bert/modeling_brand_new_bert.py`
+and `src/transformers/models/brand_new_bert/configuration_brand_new_bert.py`.
+
+Now you can finally start coding :). The generated code in `src/transformers/models/brand_new_bert/modeling_brand_new_bert.py` will
+either has the same architecture as BERT if it's an encoder-only model or BART if it's an encoder-decoder model.
+At this point, you should remind yourself what you've learned in the beginning about the theoretical aspects of the model: *How is the model different from BERT or BART?*". Implement those changes which often means to change the *self-attention* layer, the order of the normalization layer, etc...
+Here it is often useful to look at the similar architecture of already existing models in Transformers.
+
+**Note** that at this point, you don't have to be very sure that your code is fully correct or clean.
+Rather, it is advised to add a first *unclean*, copy-pasted version of the original code to 
+`src/transformers/models/brand_new_bert/modeling_brand_new_bert.py` until you feel like all the 
+necessary code is added. From our experience, it is much more efficient to quickly add a first version of the required code
+and improve/correct the code iteratively with the conversion script as described in the next section. The only thing that has to work at this 
+point is that you can instantiate the 🤗Transformers implementation of brand_new_bert, *i.e.* the following command works:
+	
+```python 
+from transformers import BrandNewBertModel, BrandNewBertConfig
+
+model = BrandNewBertModel(BrandNewBertConfig())
+```
+
+The above command will create a model according to the default parameters as defined in `BrandNewBertConfig()`
+with random weights, thus making sure that the `init()` methods of all components works.
+
+#### Write a conversion script
+
+Now, you should write a conversion script that lets you convert the checkpoint you used to debug 
+brand_new_bert in the original repository to your just created 🤗Transformers implementation of brand_new_bert.
+Here you should not try to write the conversion script from scratch, but find similar models in 🤗Transformers
+that require similar conversion scripts, *i.e.* whose original repository was written with the same framework as
+brand_new_bert.
```

Comment:
What about the case where a model can be converted to an existing architecture? Like the BORT model, there's no need to go through the last previous steps - just create a conversion script and test that the results are accurate.

Reply:
This whole paragraph is cool - but it's kind of reinventing the wheel. I would add a link to an existing conversion script and advise the user to modify it to fit their needs, rather than write it all from scratch.

Reply:
Agree, my motivation was not to re-invent the wheel, but rather explaining:

1) PyTorch class attribute names correspond to the checkpoint layer names
2) Why one should also make sure that name and size match

In the end of the paragraph it's probably better to emphasize that one should use an already written conversion script